File: hptu-score-api/src/main/java/com/hptu/score/dto/CountyAssessmentDto.java
package com.hptu.score.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import java.util.List;

public class CountyAssessmentDto {

    @NotNull
    private CountyAssessmentMetaDataDto assessmentMetaDataDto;
    @Size(min = 1)
    private List<AssessmentDto> assessments;

    public CountyAssessmentMetaDataDto getAssessmentMetaDataDto() {
        return assessmentMetaDataDto;
    }

    public void setAssessmentMetaDataDto(CountyAssessmentMetaDataDto assessmentMetaDataDto) {
        this.assessmentMetaDataDto = assessmentMetaDataDto;
    }

    public List<AssessmentDto> getAssessments() {
        return assessments;
    }

    public void setAssessments(List<AssessmentDto> assessments) {
        this.assessments = assessments;
    }
}

File: hptuScoreClient/src/app/dto/CountyAssessmentDto.ts
import { CountyAssessment } from "../models/CountyAssessment";
import { CountyAssessmentMetaData } from "../models/CountyAssessmentMetaData";

export interface CountyAssessmentDto {
    assessmentMetaDataDto: CountyAssessmentMetaData;
    assessments: CountyAssessment[];
}